switcher.py

```diff
@@ -58,16 +70,20 @@ def on_settings_changed(self, settings, keys):
         self.by_class = by_class
         screen = Wnck.Screen.get_default()
         for window in screen.get_windows():
-            instance_name = window.get_class_instance_name().lower()
+            instance_name = window.get_class_instance_name()
+            if instance_name is None:
+                continue
+            instance_name = instance_name.lower()
             if instance_name in changes:
-                self.update_window(screen, window)
+                window.connect('class-changed', self.on_window_class_changed)
```

Comment:
I'm not sure but does this overwrite the existing binding? Otherwise this would be better put elsewhere.